### PR TITLE
feat: add a query paginator

### PR DESCRIPTION
**Pull request by mutianf**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Query.java [hunk 1]
```diff
@@ -297,6 +321,75 @@ private static ByteString wrapKey(String key) {
...
+    private ByteString prevSplitPoint;
+    private boolean firstRun;
+
+    QueryPaginator(@Nonnull Query query, int chunkSize) {
+      this.originalLimit = query.builder.getRowsLimit();
+      this.newLimit = query.builder.getRowsLimit();
+      this.query = query;
+      this.chunkSize = chunkSize;
+      this.prevSplitPoint = ByteString.EMPTY;
+      this.firstRun = true;
```
[the remark this answers is not in the document]
Comment (mutianf) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Query.java:344]: hmm this should be handled by advnace and getNextQuery needs to be called after advance:

   * Query query = Query.create(...).range("a", "z");
   * Query.QueryPaginator paginator = query.createQueryPaginator(100);
   * ByteString lastSeenRowKey = ByteString.EMPTY;
   * while (paginator.advance(lastSeenRowKey)) {
   *     List<Row> rows = client.readRowsCallable().all().call(paginator.getNextQuery());
   *     for (Row row : rows) {
   *        // do some processing
   *        lastSeenRow = row;
   *     }
   * }

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Query.java [hunk 2]
```diff
@@ -297,6 +321,73 @@ private static ByteString wrapKey(String key) {
...
+  public static class QueryPaginator {
+
+    private final boolean hasOverallLimit;
+    private long remainingRows;
+    private Query query;
+    private final int pageSize;
+    private ByteString prevSplitPoint;
+
+    QueryPaginator(@Nonnull Query query, int pageSize) {
+      this.hasOverallLimit = query.builder.getRowsLimit() == 0 ? false : true;
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Query.java:338]: this.hasOverallLimit = query.builder.getRowsLimit() > 0;

**Review by igorbernstein2 — APPROVED** [review]
lgtm after nit

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Query.java (hunk 2 above)
Comment (mutianf) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Query.java:338]: 🤦‍♀️

**Pull request merged**